libs/ipc/inc/public/tfc/ipc/packet.hpp

```diff
@@ -32,7 +32,9 @@ struct header_t {
     std::copy_n(reinterpret_cast<std::byte*>(&header.type), sizeof(type), std::back_inserter(buffer));
     std::copy_n(reinterpret_cast<std::byte*>(&header.value_size), sizeof(value_size), std::back_inserter(buffer));
   }
-  static auto deserialize(header_t& result, auto&& buffer_iter) -> std::error_code {
+
+  template <typename error_t = std::error_code>
+  static auto deserialize(header_t& result, auto&& buffer_iter) -> error_t {
```

Comment:
I think this is unnecessary, and does not work for anything other than std error_code